test/jvm/generics_test.rb

```diff
@@ -44,5 +44,18 @@ def test_generics_generic_payload
 
     assert_run_output("cond string\n", cls)
   end
+
+  def test_generics_recursion
+    if JVMCompiler::JVM_VERSION.to_f >= 1.8
```

Comment:
Could you use `omit_if` here instead? Like https://github.com/mirah/mirah/blob/f2b68dab948594ebfe8cfad90e71d1ba1fbd62a2/test/jvm/interface_test.rb#L114

That way it shows up as skipped in the output instead of as a passed test.